exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetTableMetadataProviderImpl.java

```diff
@@ -109,19 +108,57 @@ public Path getSelectionRoot() {
     return selectionRoot;
   }
 
+  /**
+   * Returns list of metadata cache files
+   * @param p directory path of the cache file
+   * @param fs filesystem object
+   * @return list of cache files found in the given directory path
+   */
+  public List<Path> populateMetaPaths(Path p, DrillFileSystem fs) throws IOException {
+    List<Path> metaFilepaths = new ArrayList<>();
+    for (String filename : Metadata.CURRENT_METADATA_FILENAMES) {
+      metaFilepaths.add(new Path(p, filename));
+    }
+    for (String filename : Metadata.OLD_METADATA_FILENAMES) {
+      // Read the older version of metadata file if the current version of metadata cache files donot exist.
+      if (fileExists(fs, metaFilepaths)) {
+        return metaFilepaths;
+      }
+      metaFilepaths.clear();
+      metaFilepaths.add(new Path(p, filename));
+    }
+    if (fileExists(fs, metaFilepaths)) {
+      return metaFilepaths;
+    }
+    return new ArrayList<>();
+  }
+
+  public boolean fileExists(DrillFileSystem fs, List<Path> paths) throws IOException {
+    if (paths.isEmpty()) {
+      return true;
+    }
```

Comment:
very minor - if "paths" is empty, then the following for-loop would not iterate and "true" would be returned anyway. So this "if" statement is redundant.

Reply:
Changed